src/input/path_json.cc

```diff
@@ -25,8 +25,11 @@ PathJSON::PathJSON(istream &in)
 {
     io::filtering_istream filter_in;
 
-    filter_in.push(uncommenting_filter());
-    filter_in.push(in);
+	std::stringstream wrapped_stream;
+	wrapped_stream << "[\n" << in.rdbuf() << "\n]";
+
+	filter_in.push(uncommenting_filter());
+    filter_in.push(wrapped_stream);
```

Comment:
This wrapping should be made only when using JSON reader for reading default values. Using it here for any input stream breaks correctness of reported error paths for the user provided input file.